src/app/drive/components/ShareDialog/ShareDialog.tsx

```diff
@@ -168,6 +170,7 @@ const ShareDialog = (props: ShareDialogProps): JSX.Element => {
     setView('general');
     setIsPasswordProtected(false);
     setSharingMeta(null);
+    onCloseDialog?.();
```

Comment:
This is related to what I said above, this prop here feels weird, specially if it could be undefined, meaning that it could be not in a dialog

Reply:
This prop is more related to an action that would be executed when closing a dialog, there is no need for an action to be executed when closing same dialog in different views.